### PR TITLE
MP-9653 - Create<Entity>Command save to Localized table

### LINKED ISSUE #953: [Nox.Generator] Create<Entity>Command save to Localized table MP-9653
- relates to #907 
- depends on #936 

## Task
1. In the API Post, get the request language from the header, fall back to default language in Nox.Solution if not found, see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language, see also #937 => align with @ILoveMyTortoise since he is also implementing this part
2. Change the Create\<Entity>Command request to accept a CultureCode  
3. Update the command to:
    1. Add new translations to <Entity>Localized Table
    2. Set the translation in the Entity Fields only if is the Default Culture Code Defined in Nox

### PR DESCRIPTION
**Pull request by dejobratic**
- Extended commands with Nox.Types.CultureCode parameter;
- Added IEntityLocalizedFactory for creating localized entities based on entity and culture code; IEntityLocalizedFactory is injected in create entity command handlers if the entity has localizable fields;
- Added template for generating <Entity>LocalizedFactory for entities that have localizable fields;
- Added ClientApi tests for testing create entity commands when entity has localizable fields.

### CODE REVIEW

**Review by rochar — DISMISSED** [review]
looks good

File: src/Nox.Generator/Presentation/Api/OData/EntityController.Entity.template.cs [hunk 1]
```diff
@@ -41,6 +41,9 @@ public abstract partial class {{entity.PluralName}}ControllerBase : ODataControl
     /// </summary>
     protected readonly IMediator _mediator;
 
+    /// <symmary>
+    /// The HTTP language provider.
+    /// </symmary>
     protected readonly Nox.Presentation.Api.IHttpLanguageProvider _httpLanguageProvider;
```
Comment (gjorgji-grgovski-authoritypartners) [inline src/Nox.Generator/Presentation/Api/OData/EntityController.Entity.template.cs]:
```suggestion
    protected readonly Nox.Types.CultureCode _cultureCode ;
```

File: src/Nox.Generator/Presentation/Api/OData/EntityController.Entity.template.cs [hunk 2]
```diff
@@ -41,6 +41,9 @@ public abstract partial class {{entity.PluralName}}ControllerBase : ODataControl
...
 
     public {{entity.PluralName}}ControllerBase(
```
Comment (gjorgji-grgovski-authoritypartners) [inline src/Nox.Generator/Presentation/Api/OData/EntityController.Entity.template.cs:49]: In the constructor, we can fetch the language:
`var language = httpLanguageProvider.GetLanguage();
_cultureCode = Nox.Types.CultureCode.From(language)`

File: src/Nox.Generator/Presentation/Api/OData/EntityController.Entity.template.cs [hunk 3]
```diff
@@ -81,7 +84,8 @@ Nox.Presentation.Api.IHttpLanguageProvider httpLanguageProvider
             return BadRequest(ModelState);
         }
 
-        var createdKey = await _mediator.Send(new Create{{entity.Name}}Command({{ToLowerFirstChar entity.Name}}));
+        var language = _httpLanguageProvider.GetLanguage();
+        var createdKey = await _mediator.Send(new Create{{entity.Name}}Command({{ToLowerFirstChar entity.Name}}, Nox.Types.CultureCode.From(language)));
```
Comment (gjorgji-grgovski-authoritypartners) [inline src/Nox.Generator/Presentation/Api/OData/EntityController.Entity.template.cs]:
```suggestion
        var createdKey = await _mediator.Send(new Create{{entity.Name}}Command({{ToLowerFirstChar entity.Name}}, _cultureCode));
```

Comment (dejobratic) [reply src/Nox.Generator/Presentation/Api/OData/EntityController.Entity.template.cs]: Done. Thanks for the suggestion.

File: src/Nox.Generator/Presentation/Api/OData/EntityController.Entity.template.cs (hunk 2 above)
Comment (dejobratic) [reply src/Nox.Generator/Presentation/Api/OData/EntityController.Entity.template.cs:49]: Done. Thanks for the suggestion.

File: src/Nox.Generator/Presentation/Api/OData/EntityController.Entity.template.cs (hunk 1 above)
Comment (dejobratic) [reply src/Nox.Generator/Presentation/Api/OData/EntityController.Entity.template.cs]: Done. Thanks for the suggestion.

**Review by gjorgji-grgovski-authoritypartners — DISMISSED** [review]

**Review by gjorgji-grgovski-authoritypartners — DISMISSED** [review]

**Review by rochar — APPROVED** [review]

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

**Pull request merged**